src/cdomains/apron/sharedFunctions.apron.ml

```diff
@@ -138,10 +139,12 @@ struct
                   this query is answered by the 2 var equalities domain itself. This normalizes arbitrary expressions to a point where they
                   might be able to be represented by means of 2 var equalities *)
               let simplify e =
+                AnalysisState.executing_speculative_computations := true;
                 let ikind = try (Cilfacade.get_ikind_exp e) with Invalid_argument _ -> raise (Unsupported_CilExp Exp_not_supported)   in
                 let simp = query e ikind in
                 let const = IntDomain.IntDomTuple.to_int @@ IntDomain.IntDomTuple.cast_to ikind simp in
                 if M.tracing then M.trace "relation" "texpr1_expr_of_cil_exp/simplify: %a -> %a" d_plainexp e IntDomain.IntDomTuple.pretty simp;
+                AnalysisState.executing_speculative_computations := false;
```

Comment:
Best to wrap this in case it is ever left due to an exception, see:

https://github.com/goblint/analyzer/blob/8998e4776338305956ac6681fb751ba2a0e6bcba/src/analyses/baseInvariant.ml#L837

Also, a comment on why this is needed would probably be a good idea.